crates/ruff_dev/src/generate_docs.rs

```diff
@@ -134,13 +136,29 @@ fn process_documentation(documentation: &str, out: &mut String, rule_name: &str)
                 let anchor = option.replace('.', "_");
                 out.push_str(&format!("- [`{option}`][{option}]\n"));
                 after.push_str(&format!("[{option}]: ../settings.md#{anchor}\n"));
+                options.insert(option);
 
                 continue;
             }
         }
 
         out.push_str(line);
     }
+
+    let re = Regex::new(r"\[`([^`]*?)`]\[(.*?)]").unwrap();
+    for (_, [option, _]) in re.captures_iter(&documentation).map(|c| c.extract()) {
+        if let Some(OptionEntry::Field(field)) = Options::metadata().find(option) {
+            if !options.contains(option) {
```

Comment:
Nit: I first assumed that `options` points to `Options::metadata`, which isn't the case. What I understand is that `options` stores all linked or referenced options, maybe rename to `linked_options` or `referenced_options`?

Reply:
I renamed to `referenced_options`.